### PR TITLE
Added viewed events for quizzes and attachments

### PR DESCRIPTION
**Pull request by jlowe64**

### CODE REVIEW

**Comment by ryansmith94** [comment]
Any plans to add a UserReportEvent in here @jlowe64?

**Comment by jlowe64** [comment]
I decided against it because after looking at the actual user_report event there wasn't any extra information that I could derive. In fact, a lot of the fields were set to "NULL." It's even less information than a standard module_viewed recipe would give. If you think we should still have it separated out, then we could do that, otherwise I'd say just leave it be. 

The user_report title is a bit misleading. It's just a page with a digest of posts the user has made. The only thing you might be able to call out is "targetuser" is actually used in this event, which means if userid=whatever looked at userid=whatever1 digest, then it would have a source userid and a targetuserid, but that's all provided within the context anyway.

**Comment by ryansmith94** [comment]
Ok let's leave that out then in this repo and the Translator.

**Comment by ryansmith94** [comment]
1. Needs tests for DiscussionEvent.
3. Some of the supported events in the controller need to be documented in the readme.

File: docs/readme.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ Moodle Event | Expander Event | Test | Example
 \core\event\user_loggedout | [Event](../src/Events/Event.php) | [EventTest](../tests/EventTest.php) | [Event](examples/Event.json)
 \mod_assign\event\submission_graded | [AssignmentGraded](../src/Events/AssignmentGraded.php) | [AssignmentGradedTest](../tests/AssignmentGradedTest.php) | [AssignmentGraded](examples/AssignmentGraded.json)
 \mod_assign\event\assessable_submitted | [AssignmentSubmitted](../src/Events/AssignmentSubmitted.php) | [AssignmentSubmittedTest](../tests/AssignmentSubmittedTest.php) | [AssignmentSubmitted](examples/AssignmentSubmitted.json)
+\mod_forum\event\discussion_viewed| [DiscussionViewed](../src/Events/DiscussionEvent.php) | [ModuleEventTest](../tests/DiscussionEventTest.php) | [DiscussionViewed](examples/DiscussionEvent.json)
```
Comment (ryansmith94) [inline docs/readme.md]: `[ModuleEventTest](../tests/DiscussionEventTest.php)` needs to be `[DiscussionEventTest](../tests/DiscussionEventTest.php)`

File: docs/readme.md [hunk 2]
```diff
@@ -23,6 +23,10 @@ Moodle Event | Expander Event | Test | Example
...
+\mod_forum\event\user_report_viewed | ModuleEvent](../src/Events/ModuleEvent.php) | [ModuleEventTest](../tests/ModuleEventTest.php) | [ModuleEvent](examples/ModuleEvent.json)
```
Comment (ryansmith94) [inline docs/readme.md]: The following events need to be documented.
- '\mod_forum\event\course_module_viewed'
- '\mod_resource\event\course_module_viewed'
- '\mod_quiz\event\attempt_viewed'

**Comment by ryansmith94** [comment]
Nearly there, just a few changes to the documentation.

**Pull request merged**